src/main/java/dev/jbang/cli/BaseBuildCommand.java

```diff
@@ -125,187 +122,209 @@ void build(Script script) throws IOException {
 		// it allows integrations the options to produce the native image
 		if (!outjar.exists() || JavaUtil.javaVersion(requestedJavaVersion) < script.getBuildJdk()
 				|| nativeBuildRequired || fresh) {
-			List<String> optionList = new ArrayList<String>();
-			optionList.add(resolveInJavaHome("javac", requestedJavaVersion));
-			optionList.addAll(script.collectCompileOptions());
-			String path = script.resolveClassPath(offline);
-			if (!path.trim().isEmpty()) {
-				optionList.addAll(Arrays.asList("-classpath", path));
-			}
-			optionList.addAll(Arrays.asList("-d", tmpJarDir.getAbsolutePath()));
-
-			// add source files to compile
-			optionList.addAll(Arrays.asList(script.getBackingFile().getPath()));
-			if (script.getResolvedSources() != null) {
-				optionList.addAll(
-						script	.getResolvedSources()
-								.stream()
-								.map(x -> x.getResolvedPath().toFile().getAbsolutePath())
-								.collect(Collectors.toList()));
+			// set up temporary folder for compilation
+			Util.deleteFolder(tmpJarDir.toPath(), true);
```

Comment:
I think if it will be deleted afterwards (in `finally`), then this is probably not necessary anymore.

Reply:
Indeed, forgot to delete it after introducing that try-finally. Good catch.

Reply:
lets not assume it was successfully deleted...lots of things can go wrong - if it exists it should go. no matter what.